server/hotrod/src/main/java/org/infinispan/server/hotrod/Authentication.java

```diff
@@ -86,74 +75,109 @@ public void auth(HotRodHeader header, String mech, byte[] response) throws Privi
             server.accessLogging().logException(future, (AccessLoggingHeader) header, cause.toString(), responseBytes);
          }
       } else {
-         if (saslServer == null) {
-            ServerAuthenticationProvider sap = authenticationConfig.serverAuthenticationProvider();
-            callbackHandler = sap.getCallbackHandler(mech, authenticationConfig.mechProperties());
-            final SaslServerFactory ssf;
-            if ("EXTERNAL".equals(mech)) {
-               SslHandler sslHandler = channel.pipeline().get(SslHandler.class);
-               try {
-                  if (sslHandler != null)
-                     ssf = new ExternalSaslServerFactory(sslHandler.engine().getSession().getPeerPrincipal());
-                  else
-                     throw log.externalMechNotAllowedWithoutSSLClientCert();
-               } catch (SSLPeerUnverifiedException e) {
-                  throw log.externalMechNotAllowedWithoutSSLClientCert();
+         executor.execute(() -> authInternal(header, mech, response));
```

Comment:
While you have handled all the declared exceptions, I would still feel a bit more warm&fuzzy if any unexpected exceptions (NPEs...) would be handled with a response...